core/blockchain.go

```diff
@@ -666,6 +666,8 @@ func (self *BlockChain) InsertChain(chain types.Blocks) (int, error) {
 			PutTransactions(self.chainDb, block, block.Transactions())
 			// store the receipts
 			PutReceipts(self.chainDb, receipts)
+			// Write map map bloom filters
+			WriteMipmapBloom(self.chainDb, block.NumberU64(), receipts)
```

Comment:
it should happen during **canonical** insertion only (like the others).

We need a helper method that does a few things:

* write canon hash
* write block receipts
* write mip